app/src/main/java/com/futurice/freesound/feature/search/SearchState.java

```diff
@@ -0,0 +1,69 @@
+package com.futurice.freesound.feature.search;
+
+import com.futurice.freesound.common.utils.Preconditions;
+
+import android.support.annotation.NonNull;
+
+import polanski.option.Option;
+
+import static com.futurice.freesound.common.utils.Preconditions.get;
+
+final class SearchState {
+
+    @NonNull
+    static SearchState SEARCH_TRIGGERED() {
+        return new SearchState(true);
+    }
+
+    @NonNull
+    static SearchState SEARCH_CLEARED() {
+        return new SearchState(false);
+    }
+
+    @NonNull
+    static SearchState SEARCH_ERROR(@NonNull Throwable throwable) {
+        get(throwable);
+        return new SearchState(throwable);
+    }
+
+    private final boolean value;
+    @NonNull
+    private final Option<Throwable> optionalThrowable;
+
+    private SearchState(boolean searchTriggered) {
+        this.value = searchTriggered;
+        this.optionalThrowable = Option.none();
+    }
+
+    private SearchState(@NonNull Throwable throwable) {
+        this.value = false;
+        this.optionalThrowable = Option.ofObj(throwable);
+    }
+
+    @NonNull
+    Option<Throwable> getError() {
+        return optionalThrowable;
+    }
+
+    boolean searchTriggered() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(final Object o) {
```

Comment:
I don't think this is necessary - it gets autogenerated. (Also note that if you implement your own `equals()` you would need to also implement your own `hashCode()`